rust/repositories.bzl

```diff
@@ -1,37 +1,50 @@
 load(":known_shas.bzl", "FILE_KEY_TO_SHA")
 load(":triple_mappings.bzl", "triple_to_system", "triple_to_constraint_set", "system_to_binary_ext", "system_to_dylib_ext", "system_to_staticlib_ext")
 
-def generic_build_file(target_triple):
-    """Emits a BUILD file suitable to the provided target_triple."""
+def _sanitize_for_name(some_string):
+    """Cleans a tool name for use as a bazel workspace name"""
+
+    return some_string.replace("-", "_").replace(".", "p")
+
+def BUILD_for_compiler(target_triple):
+    """Emits a BUILD file the compiler .tar.gz."""
 
     system = triple_to_system(target_triple)
-    return """filegroup(
+    return """load("@io_bazel_rules_rust//rust:toolchain.bzl", "rust_toolchain")
+
+filegroup(
```

Comment:
opt: you could replace all those filegroup with one rust_toolchain that refer directly to the file.

Reply:
I'll hold off for now in that case... This is a change we can make in a pretty insulated way in a future PR.